libs/common/src/LocaleHelper.cpp

```diff
@@ -78,6 +78,12 @@ std::locale createUtf8Locale()
     // Don't change the locale on OSX. Using "" fails with 'locale::facet::_S_create_c_locale name not valid'
     return LocaleHelper::getBfsDefaultLocale();
 }
+#elif BOOST_OS_HAIKU
+std::locale createUtf8Locale()
+{
+    // Don't change the locale on HAIKU. Using "" fails with 'locale::facet::_S_create_c_locale name not valid'
+    return LocaleHelper::getBfsDefaultLocale();
+}
```

Comment:
This is the exact same as on MACOS, so please merge those 2 into `#elif BOOST_OS_MACOS || BOOST_OS_HAIKU`
Of course the comment needs to change too, could be split into 2 lines where the first reads `// Don't change the locale on OSX or HAIKU.`

Reply:
Done.